lib/Target.js

```diff
@@ -73,6 +77,30 @@ class Target {
     return this._pagePromise;
   }
 
+  /**
+   * @return {!Promise<?Worker>}
+   */
+  async worker() {
+    if (this._targetInfo.type !== 'service_worker' && this._targetInfo.type !== 'shared_worker')
+      return null;
+    if (!this._workerPromise) {
+      this._workerPromise = (async() => {
```

Comment:
it took me a while to understand your trick 🧐

can we make it similarly to what we do with `_pagePromise`?

```js
this._workerPromise = this._sessionFactory().then(async client => {
  // Top level workers have a fake page wrapping the actual worker.
  const [targetAttached] = await Promise.all([
    new Promise(x => client.once('Target.attachedToTarget', x)),
    client.send('Target.setAutoAttach', {autoAttach: true, waitForDebuggerOnStart: false, flatten: true}),
  ]);
  const session = Connection.fromSession(client).session(targetAttached.sessionId);
  // TODO(einbinder): Make workers send their console logs.
  return new Worker(session, this._targetInfo.url, () => {} /* consoleAPICalled */, () => {} /* exceptionThrown */);
});
```

Reply:
Done